src/celestia/celx_phase.cpp

```diff
@@ -211,15 +186,14 @@ static int phase_gc(lua_State* l)
 {
     CelxLua celx(l);
 
-    PhaseReference* ref = static_cast<PhaseReference*>(celx.checkUserData(1, Celx_Phase));
-    if (ref == nullptr)
+    auto & ref = *this_phase(l);
+    if (ref)
```

Comment:
incorrect

should be smth like
```
auto t = this_phase(l);
if (t == nullptr)
{
   auto& ref = *t;
``

Reply:
`this_phase` already checks for nullptr.

Reply:
in this case this code is too obscure. references and smart pointers pointing to nullptr is a bad practice

Reply:
Even if program should never enter this code with null pointer? Maybe `assert(t != nullptr)`?

Reply:
@375gnu So, are you sure I should do additional check in every place i get pointer to data? And should it be `if { }` statement or just `assert` would be enough?

Reply:
I'd prefer `assert`